### PR TITLE
FoldRegion: Unittest added.

### PR DESCRIPTION
**Pull request by NoreSoft**

### CODE REVIEW

File: testsuite/test-fold-region.c [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+#include <gtef/gtef.h>
+
+static GtkTextBuffer *
+test_create_and_fill_buffer (guint lines)
+{
+	GtkTextBuffer *buffer;
+	guint i;
+
+	buffer = gtk_text_buffer_new (NULL);
+	for (i = 0; i<lines; i++)
```
Comment (swilmet) [inline testsuite/test-fold-region.c]: Nitpick: space out more the code, the convention is `i < lines`. See:
https://developer.gnome.org/programming-guidelines/unstable/

File: testsuite/test-fold-region.c [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+static void
+test_destroy_last_fold_region (GtefFoldRegion *fold_region,
+			       GtkTextBuffer  *buffer)
+{
+	GtkTextTagTable *tag_table;
+
+	tag_table = gtk_text_buffer_get_tag_table (buffer);
+	g_object_unref (fold_region);
+
+	g_assert (gtk_text_tag_table_get_size (tag_table) == 0);
```
Comment (swilmet) [inline testsuite/test-fold-region.c]: The GtkTextBuffer API doesn't guarantee that the tag table is empty. GtkTextBuffer could use private tags, and if GtkSourceBuffer is used, it also uses internal tags. So this unit test should be removed.

**Comment by NoreSoft** [comment]
I deleted that test.

I also deleted the unref_buffer test, since probably no methods are meant to be called after unreffing the buffer.

**Comment by swilmet** [comment]
Pushed as commit 496ba89e4d19a12c017a47e924e6f7fada4b5485.

test-fold-region.c contained trailing spaces on a line. I already said that there should not be trailing spaces. git diff and git show show trailing spaces in red, so it is easy to see. I have a script to remove all trailing spaces, if it can help.

**Comment by swilmet** [comment]
```
#!/bin/sh
sed -i 's/[ \t]*$//' "$*"
```

**Comment by NoreSoft** [comment]
On 09.10.2016 19:33, Sébastien Wilmet wrote:
> test-fold-region.c contained trailing spaces on a line. I already said that there should not be trailing spaces. git diff and git show show trailing spaces in red, so it is easy to see. I have a script to remove all trailing spaces, if it can help.

Yes, you already said that. And of course you shouldn't have to say it
at all. I'm sorry. I don't know why I did not see it. Next time I will
use your script, to be sure.

**Pull request closed without merging**